provider/core/src/key.rs

```diff
@@ -110,6 +120,41 @@ impl Default for FallbackPriority {
     }
 }
 
+/// The string path of a data key. For example, "foo@1"
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct DataKeyPath {
+    // This string literal is wrapped in leading_tag!() and trailing_tag!() to make it detectable
+    // in a compiled binary.
+    tagged: &'static str,
+}
+
+impl DataKeyPath {
+    /// Gets the path as a static string slice.
+    #[inline]
+    pub const fn get(&self) -> &'static str {
```

Comment:
Nit: `get` is such an ambiguous name. If we want a `const` version of the deref operator, just call this `deref` as well.

Reply:
Shadowing trait functions is discouraged.

.get is used in several places in ICU4X to mean this, like DataPayload.